docs/mockwbemserver.rst

```diff
@@ -836,389 +926,310 @@ mock environment.
-created or the responder must get the set of namespaces from the CIM repository
-to generate responses to ``GetInstance``, ``EnumerateInstance``, etc.
-
-Also, the :meth:`~pywbem_mock.WBEMConnection.InvokeMethod` operation on
-on a :class:`~pywbem_mock.FakedWBEMConnection` object depends on a specific
-responder method in the mock WBEM server.
-
-To meet these needs, pywbem_mock has implemented the capability to create
+repository to generate responses (ex. ``GetInstance`` gets the instances from
+the repository, possibly filters properties and returns the instance).
+However, it is expected that the user may want WBEM operations on specific
+CIM classes to have side effects, or manage the returns differently
+than the normal responses.
+
+Thus, for example, the DMTF-defined CIM_Namespace class represents the
+namespaces in the CIM repository. That means that its responder for
+``CreateInstance`` must create a namespace as a side effect, and that its
+responders for ``GetInstance`` or ``EnumerateInstances`` must inspect the set
+of namespaces in the CIM repository to generate responses, etc.
+
+Another example of user-defined providers are classes that create their key
+property values automatically, for example classes that have a single
+``InstanceID`` key. The responder for ``CreateInstance`` of such classes would
+ignore the values for the key properties provided in the ``NewInstance``
+parameter, and determine the key values on its own.
+
+Also, the ``InvokeMethod`` operation depends on a specific responder for
+the invoked CIM method in the mock WBEM server.
+
+To meet these needs, the capability is provided to users to write
 :term:`user-defined providers <user-defined provider>` that become request
-responders for defined classes and operations which can can manipulate the
-client input and generate responses specific for defined CIM classes.
+responders for defined classes and operations.
 
-User-defined providers can be written by the user by implementing subclasses of
+User-defined providers are written by implementing subclasses of
 specific provider classes and registering these subclasses as providers using
 :meth:`~pywbem_mock.FakedWBEMConnection.register_provider`.
 
-The WBEM operations supported by user-defined providers can be implement one by
+The WBEM operations supported by user-defined providers can be implemented one by
 one. If a user-defined provider does not implement all WBEM operations, the
-default implementation is used.
+default implementation will be used.
 
 The following table shows the WBEM operations for which user-defined providers
 are supported, and the corresponding provider types:
 
-.. table:: WBEM operations of user-defined providers
+==============  ==============  ===========================================  ============================================
+WBEM operation  Provider type   Superclass                                   Description
```

Comment:
It took a bit to coordinate the refs and the header Description.  Maybe Provider Description or something would be a better word.

Reply:
DONE - changed to "Provider description" as proposed.